ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/LoadConstraint.java

```diff
@@ -124,4 +137,72 @@ public TaskTracker tasks() throws IOException, SemanticException {
     }
   }
 
+  private boolean isPrimaryKeysAlreadyLoaded(String pksMsgString) throws Exception {
+    AddPrimaryKeyMessage msg = deserializer.getAddPrimaryKeyMessage(pksMsgString);
+    List<SQLPrimaryKey> pksInMsg = msg.getPrimaryKeys();
+    if (pksInMsg.isEmpty()) {
+      return true;
+    }
+
+    String dbName = StringUtils.isBlank(dbNameToLoadIn) ? pksInMsg.get(0).getTable_db() : dbNameToLoadIn;
+    List<SQLPrimaryKey> pks;
+    try {
+      pks = context.hiveDb.getPrimaryKeyList(dbName, pksInMsg.get(0).getTable_name());
+    } catch (NoSuchObjectException e) {
+      return false;
+    }
+    return ((pks != null) && !pks.isEmpty());
+  }
+
+  private boolean isForeignKeysAlreadyLoaded(String fksMsgString) throws Exception {
+    AddForeignKeyMessage msg = deserializer.getAddForeignKeyMessage(fksMsgString);
+    List<SQLForeignKey> fksInMsg = msg.getForeignKeys();
+    if (fksInMsg.isEmpty()) {
+      return true;
+    }
+
+    String dbName = StringUtils.isBlank(dbNameToLoadIn) ? fksInMsg.get(0).getFktable_db() : dbNameToLoadIn;
+    List<SQLForeignKey> fks;
+    try {
+      fks = context.hiveDb.getForeignKeyList(dbName, fksInMsg.get(0).getFktable_name());
+    } catch (NoSuchObjectException e) {
+      return false;
+    }
+    return ((fks != null) && !fks.isEmpty());
+  }
+
+  private boolean isUniqueConstraintsAlreadyLoaded(String uksMsgString) throws Exception {
+    AddUniqueConstraintMessage msg = deserializer.getAddUniqueConstraintMessage(uksMsgString);
+    List<SQLUniqueConstraint> uksInMsg = msg.getUniqueConstraints();
+    if (uksInMsg.isEmpty()) {
+      return true;
+    }
+
+    String dbName = StringUtils.isBlank(dbNameToLoadIn) ? uksInMsg.get(0).getTable_db() : dbNameToLoadIn;
+    List<SQLUniqueConstraint> uks;
+    try {
+      uks = context.hiveDb.getUniqueConstraintList(dbName, uksInMsg.get(0).getTable_name());
+    } catch (NoSuchObjectException e) {
+      return false;
+    }
+    return ((uks != null) && !uks.isEmpty());
+  }
+
+  private boolean isNotNullConstraintsAlreadyLoaded(String nnsMsgString) throws Exception {
+    AddNotNullConstraintMessage msg = deserializer.getAddNotNullConstraintMessage(nnsMsgString);
+    List<SQLNotNullConstraint> nnsInMsg = msg.getNotNullConstraints();
+    if (nnsInMsg.isEmpty()) {
+      return true;
+    }
+
+    String dbName = StringUtils.isBlank(dbNameToLoadIn) ? nnsInMsg.get(0).getTable_db() : dbNameToLoadIn;
```

Comment:
if its a table level load ..we should use tableNameToLoadIn ?

Reply:
Table level load is not handled anywhere in bootstrap load flow. In fact I think we will get rid of it.